### PR TITLE
Separate headers pt.2: renaming namespaces

### LINKED ISSUE #453: Separate and move headers in gtclang/gridtools
`gtclang/src/gridtools` contains mixed headers. Some belong to gtclang, as they are related to gtclang DSL, like `storage_dsl.hpp`. Others are only needed by generated codes, like `storage_runtime.hpp`. Maybe there are also some which are needed by both contexts (just a doubt, didn't check).
Headers related to generated code should be moved to dawn, as we will need them to run generated code from dawn tests. Also the "shared" headers should be moved to dawn. The others should stay, but maybe a different directory structure should be chosen, as `gridtools/clang` doesn't make much sense.

### PR DESCRIPTION
**Pull request by Stagno**
## Technical Description

- Remove `gridtools::clang` namespace and substitute it with either `gtclang::dsl` for DSL or `gridtools::dawn` for gridtools-related driver includes.

- Rename `interface.hpp` to `unstructured_interface.hpp` and its namespace from `gtclang` to `dawn`.

- Move `float_type` to `dawn` namespace

- Remove `globals_impl` (not used)

- Remove `benchmark.hpp` (not used)

- Move `math.hpp`'s declarations from `driver-includes` to `gtclang_dsl_defs` (`gtclang_dsl_defs/math.hpp` namespace `gtclang::dsl::math`) and keep the definitions in dawn (`driver-includes/math.hpp` namespace `gridtools::dawn::math`). gtclang's `FunctionResolver` translates `gtclang::dsl::math` to `gridtools::dawn::math` when generating SIR.

- Remove gtclang integration tests' dependency on dawn include path, as tested gtclang DSL codes don't need to include headers in `driver-includes` anymore.

### Resolves / Enhances

resolves #453 

### Testing

Done in https://github.com/MeteoSwiss-APN/clang-gridtools/pull/142

### Dependencies

https://github.com/MeteoSwiss-APN/clang-gridtools/pull/142

### CODE REVIEW

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

File: dawn/prototype/atlasVerifyer.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #include "atlas_interface.hpp"
 #include <type_traits>
 
-// quick verifiyer class, very close in behaviour to gridtools::clang::verifyer
+// quick verifiyer class, very close in behaviour to gridtools::dawn::verifyer
```
Comment (twicki) [inline dawn/prototype/atlasVerifyer.h]: should be called Verifier - can we change it here?

File: gtclang/test/integration-test/PassPreprocessor/BracketAccessTest.cpp [hunk 2]
```diff
@@ -42,7 +41,7 @@ stencil Test01 {
 };
 
 stencil Test02 {
-  gridtools::clang::storage foo, bar;
+  storage foo, bar;
```
Comment (twicki) [inline gtclang/test/integration-test/PassPreprocessor/BracketAccessTest.cpp]: what this not part of the test?

File: gtclang/test/integration-test/PassPreprocessor/BracketAccessTest.cpp [hunk 3]
```diff
@@ -51,7 +50,7 @@ stencil Test02 {
 };
 
 stencil Test03 {
-  clang::storage foo, bar;
+  storage foo, bar;
```
Comment (twicki) [inline gtclang/test/integration-test/PassPreprocessor/BracketAccessTest.cpp]: also this

**Pull request merged**